packages/starlight/utils/navigation.ts

```diff
@@ -341,4 +355,16 @@ function applyPrevNextLinkConfig(
 }
 
 /** Remove the extension from a path. */
-const stripExtension = (path: string) => path.replace(/\.\w+$/, '');
+export function stripExtension(path: string) {
+	path = stripTrailingSlash(path);
+	return path ? path.replace(/\.\w+$/, '') + '/' : '/';
+}
+
+/** Add '.html' extension to a path. */
+export function ensureHtmlExtension(path: string) {
+	path = stripTrailingSlash(path);
+	if (path.endsWith('.html')) return ensureLeadingSlash(path);
+
+	path = stripLeadingAndTrailingSlashes(path);
+	return path ? ensureLeadingSlash(path) + '.html' : '/index.html';
+}
```

Comment:
What do you think of moving these to `utils/path.ts`? When they were only used by the navigation logic, keeping them in this file made sense, but now we’re sharing them also in `utils/format-path.ts`, I think that might be the more correct place for them